### DISCUSSION TITLE
keyball39: TRSケーブルを繋いで時に片方が動作しない。トラックボールが動作しない。

### DISCUSSION THREAD

**Discussion by TKBxcl32** [Yowkees/keyball#221] [Q&A]
お世話になります。
表題の件についての質問になります。
1.TRSケーブルを接続した際に場合に、USBを繋いでいない片方の入力が行われません。片方のみをUSBケーブルで接続した場合は両方とも動作いたします。回路図を見てもあまり理解ができず不良箇所を特定する際にはどこを調べればいいでしょうか？
![IMG_1550 - コピー](https://user-images.githubusercontent.com/39870325/233074758-5d63ae86-556e-441d-b019-ea07d8f8e83e.JPG)
![IMG_1551 - コピー](https://user-images.githubusercontent.com/39870325/233074786-7e3cec18-8f27-49cf-aae5-b43355dafc42.JPG)

2.トラックボールが動作いたしません。全く動かない状態です。センサー側のピン先を一つはんだ付けする際に折ってしまいまして、見たところ繋がっているようには見えるのですがそれらが原因かと考えています。こちらの基板とパ－ツは購入可能でしょうか？

![IMG_1546](https://user-images.githubusercontent.com/39870325/233074799-7ed755bc-3d59-479a-9a25-8988eb39745d.jpg)

こちらでも何度か、TRSコネクタやpromicroのピンのはんだ付けし直し等を行いましたが、動作の改善ができなかったため。相談させていただきました。またはんだ付け初心者ですのでここがだめ等あればご教授いただきたいです。
お手数おかけいたしますが。ご回答いただけますと幸いです。

**Comment by Yowkees** [comment]
![KB39RTRS](https://user-images.githubusercontent.com/84209772/233255205-2624f44e-d33b-4a52-857a-688a28e2364f.jpg)
![KB39LTRS](https://user-images.githubusercontent.com/84209772/233255220-b248d95d-91ac-4c42-8576-bb90fae61c03.jpg)
ご連絡ありがとうございます。
左右通信が動作しない場合、ProMicro本体、もしくはProMicro裏の半田付け、TRSコネクタの半田付けが原因である場合がおおいですが、お写真を見るとそれほど問題なさそうですね。
念のため赤丸部を見直してください。
ProMicro裏は隣の端子とショートしていないか、テスター等でご確認ください。
TRSコネクタはちょっと半田が少ない気がしました。基板の銀パッドと半田がなじんで、山型になるようにしてください。

また、稀にTRSケーブルが悪い場合があります。ケーブルを他のものに変えて試していただけませんでしょうか。

トラックボール基板や光学センサーを販売可能です。配送先等お話したいので、白銀ラボウェブサイトの問合せよりメールをいただけますでしょうか。https://shirogane-lab.com/pages/contact

お手数ですがご対応お願いいたします。

**Reply by TKBxcl32** [reply]
ご返答ありがとうございます。本日ソケットを外して確認したところ下記写真の囲ってある部分がショートしているようです。はんだ吸い取り線等でもなかなか取れないのですが、対処法等はありますでしょうか？
![IMG_1552 - コピー](https://user-images.githubusercontent.com/39870325/233397657-2e75d6a7-e97e-4900-9b0b-5134fbed0946.jpg)

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。
ご連絡いただいた３点はすべてGNDですので、ショートしていて大丈夫です。
もう一つ上のスルーホールが左右通信用のDATA端子なので、そことショートしていないかと思いまして。

よろしければ、下記にKeyball39の基板データを公開していますので、KiCADで表示いただければ参考になると思います。
https://github.com/Yowkees/keyball/tree/main/keyball39/design_data

お手数ですがご対応お願いいたします。

**Reply by TKBxcl32** [reply]
こちらの問題、先程すべての動作が正常に動作すること確認できました。

原因はピンソケットが変形していたことでした。

おそらくpromicro側のピンを指す過程で無理やり合わせようとしたときに穴が変形したものと思われます。

ピンソケットを別のものに変えることで正常に動作することができました。
今回は対応いただきありがとうございました。
